pkg/runtime/serializer/codec_factory.go

```diff
@@ -73,29 +142,24 @@ func newCodecFactory(scheme *runtime.Scheme, mf json.MetaFactory) CodecFactory {
 			}
 			alreadyAccepted[mediaType] = struct{}{}
 			accepts = append(accepts, mediaType)
+			if mediaType == "application/json" {
+				legacySerializer = d.Serializer
+			}
 		}
 	}
+	if legacySerializer == nil {
+		legacySerializer = serializers[0].Serializer
```

Comment:
It's not obvious to my why this is better than leaving it `nil`.  If not every `CodecFactory` has a `legacySerializer`, that seems like an ok state.